src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/resource/VisitResourceIntTest.kt

```diff
@@ -76,15 +78,19 @@ class VisitResourceIntTest : IntegrationTestBase() {
     offenderBookingId = offender.latestBooking().bookingId
   }
 
-  @AfterEach
-  internal fun deletePrisoner() {
-    repository.delete(offender)
-    repository.delete(threePeople)
-  }
-
   @DisplayName("Create")
   @Nested
   inner class CreateVisitRequestTest {
+    @BeforeEach
+    internal fun setUpData() {
+      createPrisoner()
+    }
+
+    @AfterEach
+    internal fun deleteData() {
+      repository.delete(offender)
+      repository.delete(threePeople)
+    }
 
```

Comment:
moving data requirements into nested classes to allow specific test data creation